plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+      output_string = []
+      for token in tokens:
+        if token not in self._DELIMITERS:
+          token = obfuscated_string_map.get(token, token)
+        output_string.append(token)
+      output_strings.append(''.join(output_string))
+
+    return output_strings
+
+  def _DecryptStrings(self, extracted_strings, aes_key):
+    """Decrypt strings extracted from OneDrive logs.
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and the initialization vector is null.
+    * URL safe base64 encoded.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        token_length = len(token)
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if token_length < 22:
+          output_string.append(token)
+          continue
+
+        if not all(character in self._BASE64_ALPHABET for character in token):
+          output_string.append(token)
+          continue
+
+        rem = token_length % 4
+        if rem == 1:
+          output_string.append(token)
+          continue
+
+        while len(token) % 4:
+          token += '='
+
+        ciphertext = base64.urlsafe_b64decode(token)
+
+        if len(ciphertext) % 16:
+          output_string.append(token)
+          continue
+
+        aes_context = pycaes.context()
+        aes_context.set_key(pycaes.crypt_modes.DECRYPT, aes_key)
+        plaintext = pycaes.crypt_cbc(
+            aes_context, pycaes.crypt_modes.DECRYPT,
+            self._AES_IV, ciphertext)
+
+        unpadder = padding.PKCS7(128).unpadder()
+        try:
+          data = unpadder.update(plaintext) + unpadder.finalize()
+        except ValueError:
+          output_string.append(token)
+          continue
+
+        try:
+          output_string.append(data.decode('utf-16-le'))
```

Comment:
can you explain the logic here, why is the data considered correct if decoding fails?

Reply:
also maybe move `output_string.append()` out of the try-except

Reply:
Fixed to append the original token if decoding fails.

Done.